.github/workflows/client.yml

```diff
@@ -12,22 +12,50 @@ on:
       - "solidity/dashboard/**"
   pull_request:
   workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment for workflow execution'
+        required: false
+        default: 'dev'
+      upstream_builds:
+        description: 'Upstream builds'
+        required: false
+      ref:
+        description: 'Git reference to checkout (e.g. branch name)'
+        required: false
+        default: 'master'
 
 jobs:
   client-detect-changes:
     runs-on: ubuntu-latest
     outputs:
       path-filter: ${{ steps.filter.outputs.path-filter }}
     steps:
-    - uses: actions/checkout@v2
-      if: github.event_name == 'pull_request' 
-    - uses: dorny/paths-filter@v2
-      if: github.event_name == 'pull_request' 
-      id: filter
-      with:
-        filters: |
-          path-filter:
-            - './!((docs|infrastructure|scripts|token-stakedrop|(solidity/dashboard))/**)'
+      - name: Show workflow metadata
+        if: github.event_name == 'workflow_dispatch'
+        run: |
+            echo "Workflow dispatched on branch: ${{ github.ref }}"
+            echo "environment = ${{ github.event.inputs.environment }}"
+            echo "upstream_builds = ${{ github.event.inputs.upstream_builds }}"
+            echo "ref = ${{ github.event.inputs.ref }}"
+            echo "sha = ${{ github.sha }}"
+
+      - name: Show workflow metadata
+        if: github.event_name != 'workflow_dispatch'
+        run: |
+            echo "ref = ${{ github.ref }}"
+            echo "sha = ${{ github.sha }}"
```

Comment:
Weren't we using these just for debugging? Do we want to keep them?

Reply:
Initially I introduced them for debugging purposes during feature development, but now I'm thinking of keeping them, as they provide better visibility of why the workflow was started and what parameters were used (GitHub Actions don't list the workflow's dispatch parameters anywhere afaik).

Reply:
Removed in ce1219c404fecaeecc49904dfe79cfc552352f3d